micropip/freeze.py

```diff
@@ -20,11 +19,13 @@ def freeze_data(
     lockfile_packages: dict[str, dict[str, Any]], lockfile_info: dict[str, str]
 ) -> dict[str, Any]:
     pyodide_packages = deepcopy(lockfile_packages)
-    pip_packages = load_pip_packages()
-    package_items = itertools.chain(pyodide_packages.items(), pip_packages)
+    packages = dict(load_pip_packages())
+    packages.update(
+        {name: info for name, info in pyodide_packages.items() if name not in packages}
```

Comment:
Minor improvement:
```suggestion
        [name, info] for name, info in pyodide_packages.items() if name not in packages
```

Reply:
I haven't added any tests: any thoughts? Is there a running HTTP server during tests that it could pull from? Maybe a file system one (should still be picked up in `pip_packages`?